### PR TITLE
Add the component for the homepage grid

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Related to #1433 by @panchovm.

## Description

This PR adds the gallery grid component shown on the redesigned home page. Care has been taken to minimise disruption to the existing homepage.

## Testing Instructions

0. Check out the branch and open Storybook.
1. Explore the story for `VHomeGallery`.
2. Compare the images with the mockups (including position).
3. Check their links and titles.
4. Resize width to see if the column count is updated.
5. Check animation with and without reduced motion.

---

0. Check out the branch and run the dev server.
1. Visit the home page with the `new_header` flag enabled (`http://localhost:8443/?ff_new_header=on`).
2. Observe the new home page.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
The gallery looks awesome ✨ a few notes:

## Focus style

The square ring looks odd wrapping a circle element.

CleanShot 2022-12-21 at 18 11 39@2x

Based on [this comment](https://github.com/WordPress/openverse-frontend/issues/2049#issuecomment-1358044819) on focus ring styles, it would be great to apply the `Filled Bold` version. Here is a quick mockup of how it would look like.



## Alt text

Still pending the alt text for all images. Added to my tasks.

## Vertical alignment

Not sure if this comment belong to this PR, yet sharing it to not miss it. The vertical center alignment of search area and gallery is not even. I think `h2` has a top margin spacing adding extra space.

CleanShot 2022-12-21 at 18 46 46@2x

## Broken link

I found just one: Universe Image-06.

---

Despite the above, it looks awesome. Great work.

File: src/components/VHomeGallery/VHomeGallery.vue
Comment (obulat) [inline src/components/VHomeGallery/VHomeGallery.vue:92]:
```suggestion
```

Comment (obulat) [reply src/components/VHomeGallery/VHomeGallery.vue:92]: Left-over logging

**Review by obulat — APPROVED** [review]
Looks and works fantastic 🌟  You implemented it so fast 🚀 
I left one suggestion for removing logging.

File: src/components/VHomeGallery/VHomeGallery.vue
Comment (dhruvkb) [reply src/components/VHomeGallery/VHomeGallery.vue]: Ah I always forget! I should add a lint rule for this. Removed in 302f703.

Comment (obulat) [reply src/components/VHomeGallery/VHomeGallery.vue]: I do too :D

**Comment by panchovm** [comment]
### Delay in focus when jumping
When jumping between gallery images in dev using the keyboard, the focus loads its style with a small delay, showing the pink ring first and the white ring later. Here is a screen recording.



Not sure if this comment belongs here but adding it as I also shared [feedback on how focus ring styles work](https://github.com/WordPress/openverse-frontend/issues/2049#issuecomment-1365247066).

All the rest looks excellent ✨ 

---

# Non-blocking idea
### Layout in `> 2xl`

I am reviewing from a `2300px` display, and the layout has a big white space in the middle.

image

I was trying some ideas, and putting the content in the center looks good. The search area and gallery widths are the same and placed in the center with a gap of `200px` in between. The heading, sentences, and search bar width are `100%`. It might look like this. 

image

**Comment by dhruvkb** [comment]
@panchovm I did not experience the behaviour seen by you. On my monitor, with width of 2560px, this is the appearance of the gallery. There is white space but the gallery is centered in the right field.

image

**Comment by panchovm** [comment]
Oh, I see.

Nonetheless, in both cases, the side and in-between spaces look uneven. Here is a wireframe of the idea.



It is an idea that should not block the work made, so please feel free to avoid it. We can tackle it in the future.

**Comment by dhruvkb** [comment]
@panchovm the diagram you show should be from 2xl breakpoint and above?

**Comment by panchovm** [comment]
Yes! Sorry for not being clear @dhruvkb

**Comment by obulat** [comment]
Can we merge this PR as is, and open a separate issue for the 2xl fixes, @panchovm , @dhruvkb ? It would be nice to unblock other homepage development.

**Comment by dhruvkb** [comment]
I'm more than happy to get this merged and fix the 2xl later.

**Comment by obulat** [comment]
> I'm more than happy to get this merged and fix the 2xl later.

> It is an idea that should not block the work made, so please feel free to avoid it. We can tackle it in the future.

I think we can take it as the second approval :) Considering that these changes are only active when the new_header is on, I believe it's safe to merge after the test failures are fixed.

**Comment by panchovm** [comment]
Sure. No problem merging it.

**Review by panchovm — APPROVED** [review]

File: src/pages/index.vue
Comment (obulat) [inline src/pages/index.vue:4]: Can we remove the header and footer on the index page from this PR, leaving just the addition of the HomeGallery? Now that all pages have the same header and footer, we can simplify the layouts: use the `default` layout for all pages (including index and 404).  I was hoping to open a PR for moving `index.vue` and 404 pages to `default` layout after this PR is merged.

File: test/playwright/visual-regression/pages/homepage.spec.ts
Comment (obulat) [inline test/playwright/visual-regression/pages/homepage.spec.ts:52]: If we add something like "old design" to the other test, it will be easier to clean up after the full transition to the new designs.

Comment (dhruvkb) [reply test/playwright/visual-regression/pages/homepage.spec.ts:52]: Makes sense, updated test names.

**Pull request merged**
